src/main/java/se/sundsvall/smloader/service/OpenEService.java

```diff
@@ -0,0 +1,101 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import se.sundsvall.smloader.integration.db.CaseRepository;
+import se.sundsvall.smloader.integration.openeexternal.OpenEExternalClient;
+import se.sundsvall.smloader.integration.openeexternalsoap.OpenEExternalSoapClient;
+import se.sundsvall.smloader.integration.openeinternal.OpenEInternalClient;
+import se.sundsvall.smloader.integration.openeinternalsoap.OpenEInternalSoapClient;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+import static se.sundsvall.smloader.integration.util.XPathUtil.evaluateXPath;
+import static se.sundsvall.smloader.service.mapper.CaseMapper.toCaseEntity;
+
+@Service
+public class OpenEService {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(OpenEService.class);
+
+	@Value("${integration.open-e-external.family-ids:}")
+	private String externalFamilyIds;
+
+	@Value("${integration.open-e-internal.family-ids:}")
+	private String internalFamilyIds;
+
+	private final OpenEExternalClient openEExternalClient;
+	private final OpenEInternalClient openEInternalClient;
+	private final OpenEExternalSoapClient openEExternalSoapClient;
+	private final OpenEInternalSoapClient openEInternalSoapClient;
+	private final CaseRepository caseRepository;
+
+	public OpenEService(OpenEExternalClient openEExternalClient, OpenEInternalClient openEInternalClient, OpenEExternalSoapClient openEExternalSoapClient, OpenEInternalSoapClient openEInternalSoapClient,
+			CaseRepository caseRepository) {
+		this.openEExternalClient = openEExternalClient;
+		this.openEInternalClient = openEInternalClient;
+		this.openEExternalSoapClient = openEExternalSoapClient;
+		this.openEInternalSoapClient = openEInternalSoapClient;
+		this.caseRepository = caseRepository;
+	}
+
+	public void fetchAndSaveNewOpenECases(String status, LocalDateTime fromDate, LocalDateTime toDate) {
+		handleExternalCases(status, fromDate, toDate);
+
+		handleInternalCases(status, fromDate, toDate);
+	}
+
+	private void handleExternalCases(String status, LocalDateTime fromDate, LocalDateTime toDate) {
+
+		final var externalFlowInstanceIds = Arrays.stream(this.externalFamilyIds.split(","))
+			.map(familyId -> openEExternalClient.getErrandIds(familyId, status, fromDate.toString(), toDate.toString()))
+			.map(this::getErrandIds)
+			.flatMap(List::stream)
+			.distinct()
+			.toList();
+
+		externalFlowInstanceIds.forEach(flowInstanceId -> {
+			if (caseRepository.existsById(flowInstanceId)) {
+				LOGGER.info("Case with id: '{}' already exists in database. Nothing will be saved.", flowInstanceId);
+				return;
+			}
+			final var openECase = openEExternalClient.getErrand(flowInstanceId);
+			if (nonNull(openECase)) {
+				caseRepository.save(toCaseEntity(flowInstanceId, openECase));
+			}
+		});
+	}
```

Comment:
A flowInstanceId is only a number with like 6 digits. Now when we have two instances of open-e, we will have to keep track of which cases are internal and external.

At row 63, we should probably use: "existsByFlowInstanceIdAndInstance".

For example, if we start by fetching all externalcases and persisting those. We might have some conflicts when we fetch all internalcases because the flowinstanceid is already in the database.